plugins/catalog-backend-module-gitlab/src/lib/types.ts

```diff
@@ -136,3 +142,36 @@ export type GitlabProviderConfig = {
   schedule?: TaskScheduleDefinition;
   skipForkedRepos?: boolean;
 };
+
+/**
+ * Customize how group names are generated
+ *
+ * @public
+ */
+export type GroupNameTransformer = (
```

Comment:
can we make all these functions have a single object argument instead? for example:

```ts
export type GroupNameTransformer = (options: {
  group: GitLabGroup,
  providerConfig: GitlabProviderConfig,
})
```

with carefully chosen names for the keys that are maximally informative for the user :) feel free to use `providerConfig` instead of `provConfig` etc

doing this makes the call sites much clearer, you get nice completion, you can introduce optional arguments more easily, it extends better in the future etc

Reply:
Of course - initially I hoped to need no other information that the GitLabGroup... will improve the variable names!

Reply:
As I need the same options twice, in my types and in my default callback, I created an interface for them - ok for you?